llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1475,36 +1438,74 @@ void RISCVInsertVSETVLI::doLocalPostpass(MachineBasicBlock &MBB) {
 
     if (!isVectorConfigInstr(MI)) {
       doUnion(Used, getDemanded(MI, MRI));
+      // We can't handle the case when the source AVL
+      // register of *NextMI is defined after MI
+      if (NextMI && NextMI->getOperand(1).isReg() && MI.getOperand(0).isReg() &&
```

Comment:
MI here can be any instruction right? What if it has more than 1 def?

Reply:
Or it might not even be a def. It could be a store instruction that uses the same register?

Reply:
To address this, I've tried to make the check as general as possible: https://github.com/llvm/llvm-project/pull/67144/commits/22d8abb63a7ef94b34109add4b20949fd682fbf2.